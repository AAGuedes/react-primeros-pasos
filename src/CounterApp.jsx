import { useState } from 'react'
import PropTypes from "prop-types";



export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const counterAdd = () => setCounter(counter + 1);
    const counterSubstract = () => setCounter(counter - 1);
    const counterReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={counterAdd}>
                +1
            </button>
            <button onClick={counterSubstract}>
                -1
            </button>
            <button aria-label="btn-reset" onClick={counterReset}>
                Reset
            </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number,
}
