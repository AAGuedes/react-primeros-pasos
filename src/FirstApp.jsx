import PropTypes from 'prop-types'

const newMessage = {
    message: 'Hello World!',
    title: 'FirstApp',
}

const getResult = (a, b) => {
    return a + b;
}

// Lo que está englobado por <></> puede hacerse también con Fragment
export const FirstApp = ({ title, subtitle, name }) => {

    // console.log(props);

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    subtitle: PropTypes.string,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'John Doe',
    subtitle: 'No subtitle',
}
