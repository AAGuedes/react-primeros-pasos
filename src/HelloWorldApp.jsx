// function App() {
//     return <h1>Hello World!</h1>;
// }

// export default App;

// Creado con rafc
export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  )
}
