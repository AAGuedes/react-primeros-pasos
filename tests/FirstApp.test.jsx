import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas de <FirstApp />', () => {

    // test('Debe de coincidir con el Snapshot', () => {

    //     const title = `Hello, I'm John Doe`;
    //     const { container } = render(<FirstApp title={title} />);

    //     expect(container).toMatchSnapshot();

    // });

    test('Debe mostrar el tÃ­tulo en un h1', () => {

        const title = `Hello, I'm John Doe`;
        const { container, getByText, getByTestId } = render(<FirstApp title={title} />);

        expect(getByText(title)).toBeTruthy();

        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain(title);

        expect(getByTestId('test-title').innerHTML).toContain(title);

    });

    test('Debe mostrar el subtitulo enviado por props', () => {

        const title = `Hello, I'm John Doe`;
        const subtitle = `No subtitle`;
        const { getAllByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        expect(getAllByText(subtitle).length).toBe(2);

    });

});
