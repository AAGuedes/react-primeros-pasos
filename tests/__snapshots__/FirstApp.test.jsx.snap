// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <FirstApp /> Debe de coincidir con el Snapshot 1`] = `
<div>
  <h1>
    Hello, I'm John Doe
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    John Doe
  </p>
</div>
`;
